src/components/Upload/UploadFile.vue

```diff
@@ -1,30 +1,52 @@
 <template>
-	<div
-		class="card-body flex justify-center border-3 border-gray-400 border-dashed 40vh"
-	>
-		<div class="flex flex-col justify-center items-center">
-			<i class="fa fa-cloud-upload-alt my-8 text-gray-400 text-6xl"></i>
-			<h3 class="my-1 text-md leading-5 text-gray-800">
-				Pretiahnite súbory sem
-			</h3>
-			<p class="my-1 text-md text-gray-600">alebo</p>
-			<div class="mx-auto my-4">
-				<label
-					class="flex text-white rounded bg-primary-500 hover:bg-primary-400 px-4 py-2 cursor-pointer transition duration-500 ease select-none focus:outline-none"
-				>
-					Vybrať súbory
-					<input
-						id="file"
-						ref="file"
-						type="file"
-						multiple
-						class="hidden"
-						@change="handleFileUpload"
-					/>
-				</label>
-				<button v-on:click="submitFile">Submit</button>
+	<div class="space-y-6">
+		<div class="card-body border-3 border-gray-400 border-dashed 40vh">
+			<div class="card-body flex flex-col justify-center bg-primary-100 hover:bg-white">
+				<form ref="fileform" class="cursor-pointer">
+					<div class="flex flex-col justify-center items-center space-y-2 py-8">
+						<i class="fa fa-cloud-upload-alt text-gray-500 text-6xl"></i>
+						<h2 class="text-xl font-semibold leading-6 text-gray-800">
+							Pretiahnite súbory sem
+						</h2>
+						<p class="text-md text-gray-600">alebo</p>
+						<div class="mx-auto my-4">
+							<label
+								class="flex text-white rounded shadow-md bg-primary-500 hover:bg-primary-400 px-6 py-2 
+								cursor-pointer transition duration-500 ease select-none focus:outline-none"
+							>
+								Vybrať súbory
+								<input
+									id="files"
+									ref="files"
+									type="file"
+									multiple
+									class="hidden"
+									@change="handleFileUpload"
+								/>
+							</label>
+						</div>				
+					</div>
+				</form>
 			</div>
 		</div>
+		
+		<div v-show="this.$parent.files.length > 0" class="overflow-auto h-40 bg-primary-100 shadow-md">
+			<div v-for="(file, key) in this.$parent.files" 
+				:key="key" 
+				class="flex justify-between items-center px-4 my-2 border-b-3 border-gray-400"
+			>				
+				<div class="flex items-center space-x-4">
+					<i class="fas fa-file text-gray-500 text-3xl hover:text-gray-600"></i>
+					<p> {{ file.name }} </p>					
+				</div>
+				<div>
+					<button class="p-2 flex justify-center items-center focus:outline-none" v-on:click="removeFile( key )">
+						<i class="fas fa-trash text-gray-500 text-2xl hover:text-gray-600"></i>					
```

Comment:
Can we use group hover, which would add this icon the text, and also add slight hover background on the `button` element?

Reply:
I don't get it 🙈

Reply:
https://tailwindcss.com/docs/hover-focus-and-other-states#group-hover

Group hover.

Now, if you hover over the `button`, nothing changes, even though you can click on it, the icon color is still the same.

You need to hover over the `i` element to change the color of the icon.
What I want from you is to make group-hover to work, so you hover over the button, it would add slightly darker background to the button, and also would add text to the icon.


You can do it either (thanks to color being propagated to children):
```html
<button class="bg-white hover:bg-gray-100 hover:text-gray-600>
 <i class="... text-gray-500"
<button />
```
or 
```html
<button class="group bg-white hover:bg-gray-100">
 <i class="... text-gray-500 group-hover:text-gray-600"
</button>
```
The second one would need some changes in tailwind I think.  You can decide between those two.
Group hover is used when you need to add style to the children element, while hovering over its ancestor.